app/controllers/previousRegistrations/PreviousOssNumberController.scala

```diff
@@ -47,69 +47,71 @@ class PreviousOssNumberController @Inject()(
-            case countryWithValidationDetails =>
-              Future.successful(Ok(view(preparedForm, waypoints, countryIndex, schemeIndex, countryWithValidationDetails, previousSchemeHintText)))
-          }
-      }
-  }
+  def onPageLoad(waypoints: Waypoints, countryIndex: Index, schemeIndex: Index): Action[AnyContent] =
+    cc.authAndGetData(waypoints.registrationModificationMode).async {
+      implicit request =>
+        getPreviousCountry(waypoints, countryIndex) {
+          country =>
 
-  def onSubmit(waypoints: Waypoints, countryIndex: Index, schemeIndex: Index): Action[AnyContent] = cc.authAndGetData(waypoints.inAmend).async {
-    implicit request =>
-      getPreviousCountry(waypoints, countryIndex) {
-        country =>
-
-          val previousSchemeHintText = determinePreviousSchemeHintText(countryIndex)
-
-          val form = request.userAnswers.get(AllPreviousSchemesForCountryWithOptionalVatNumberQuery(countryIndex)) match {
-            case Some(previousSchemeDetails) =>
-              val previousSchemes = previousSchemeDetails.flatMap(_.previousScheme)
-              formProvider(country, previousSchemes)
-
-            case None =>
-              formProvider(country, Seq.empty)
-          }
-
-          form.bindFromRequest().fold(
-            formWithErrors =>
-              CountryWithValidationDetails.euCountriesWithVRNValidationRules.filter(_.country.code == country.code).head match {
-                case countryWithValidationDetails =>
-                  Future.successful(BadRequest(view(
-                    formWithErrors, waypoints, countryIndex, schemeIndex, countryWithValidationDetails, previousSchemeHintText)))
-              },
-
-            value => {
-              val previousScheme = if (value.startsWith("EU")) {
-                PreviousScheme.OSSNU
-              } else {
-                PreviousScheme.OSSU
-              }
-              searchSchemeThenSaveAndRedirect(waypoints, countryIndex, schemeIndex, country, value, previousScheme)
+            val previousSchemeHintText = determinePreviousSchemeHintText(countryIndex)
+
+            val form = request.userAnswers.get(AllPreviousSchemesForCountryWithOptionalVatNumberQuery(countryIndex)) match {
+              case Some(previousSchemeDetails) =>
+
+                val previousSchemes = previousSchemeDetails.flatMap(_.previousScheme)
+                formProvider(country, previousSchemes)
+
+              case None =>
+                formProvider(country, Seq.empty)
+            }
+
+            val preparedForm = request.userAnswers.get(PreviousOssNumberPage(countryIndex, schemeIndex)) match {
+              case None => form
+              case Some(value) => form.fill(value.previousSchemeNumber)
+            }
+            CountryWithValidationDetails.euCountriesWithVRNValidationRules.filter(_.country.code == country.code).head match {
```

Comment:
Could use `find` to avoid calling `head`: 
```
CountryWithValidationDetails.euCountriesWithVRNValidationRules.find(_.country.code == country.code) match {
  case Some(countryWithValidationDetails) =>
```